### PR TITLE
Fixes for MainMenuMessages

### PR DESCRIPTION
**Pull request by zorgesho**
- Fixed an exception that prevented proper deinitialization
- Fixed conflict with HudConfig mod
- Fixed patch for BelowZero (the same transpiler works for both games now)
- Minor refactoring

### CODE REVIEW

File: QModManager/Utility/MainMenuMessages.cs [hunk 1]
```diff
@@ -121,98 +137,105 @@ static IEnumerator _waitForLoad()
...
-
-                list.InsertRange(index, new List<CodeInstruction>
+                try
                 {
-                    new CodeInstruction(OpCodes.Ldloc_S, 6),
-                    new CodeInstruction(OpCodes.Call, AccessTools.Method(typeof(Patches), nameof(_getVal)))
-                });
-
-                return list;
+                    var code = new CodeMatcher(cins).
```
Comment (PrimeSonic) [inline QModManager/Utility/MainMenuMessages.cs:217]: Not many of us have used `CodeMatcher` before.  
Could you add some more comments explaining how this all works?

Comment (zorgesho) [reply QModManager/Utility/MainMenuMessages.cs:217]: Hmm, I'm not sure where to put comments. Basically, it's just an advanced wrapper around searching in the list of instructions in Traverse/LINQ style. It stores current instruction index, so there is no need to create and check variables for indexes and such.
So in this case it searches for specified method from the end, stores index from the operand of the finded instruction and then continues to search from the same position and then inserts couple of instructions.
It's almost the same transpiler as before, but SN and BZ have slightly different versions of that method and local indexes are not the same for that variable.

Comment (PrimeSonic) [reply QModManager/Utility/MainMenuMessages.cs:217]: Just some comments explaining what these methods are doing: `MatchBack`, `Advance`, `MatchForward`, `EmitDelegate`

Comment (zorgesho) [reply QModManager/Utility/MainMenuMessages.cs:217]: Well, `CodeMatcher` have its own documentation that is accessible in Visual Studio. And as for github, isn't those method names are straightforward enough to get general idea of what is going on?

Comment (PrimeSonic) [reply QModManager/Utility/MainMenuMessages.cs:217]: I honestly find the fluent syntax of `CodeMatcher` to be harder to follow that a hand-crafted state machine doing the same thing.

We've gone back and forth enough here. We can continue this on Discord.

**Review by PrimeSonic — APPROVED** [review]
I can get behind this with the added comments

**Pull request merged**